File: sgf-poker/sgf-poker-ios/Core/UseCases/DeleteGameUseCase.swift
// Core/UseCases/DeleteGameUseCase.swift

import Foundation

/// Removes a game from the repository by its identifier.
final class DeleteGameUseCase {

    private let repository: GameRepository

    init(repository: GameRepository) {
        self.repository = repository
    }

    /// - Parameter id: The UUID of the game to remove.
    /// - Throws: `GameError.gameNotFound` if no matching game exists.
    func execute(id: UUID) throws {
        try repository.delete(id: id)
    }
}

File: sgf-poker-android/app/src/main/java/com/sgf/poker/usecases/game/DeleteGameUseCase.java
package com.sgf.poker.usecases.game;

import com.sgf.poker.data.repository.GameRepository;

/** Removes a game by id. */
public class DeleteGameUseCase {

    private final GameRepository repository;

    public DeleteGameUseCase(GameRepository repository) {
        this.repository = repository;
    }

    public void execute(String gameId) {
        repository.delete(gameId);
    }
}
